pkg/controllers/multicluster/utils.go

```diff
@@ -177,7 +216,7 @@ func CreateDerivedServiceStruct(svcImport *multiclusterv1alpha1.ServiceImport, i
 	return &v1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace:       svcImport.Namespace,
-			Name:            svcImport.Annotations[DerivedServiceAnnotation],
+			Name:            DerivedName(svcImport.Namespace, svcImport.Name, clusterId),
 			OwnerReferences: []metav1.OwnerReference{*ownerRef},
```

Comment:
Please add `multicluster.kubernetes.io/source-cluster` label to derived service.

Reply:
KEP does not specify this - but I do like the idea of the Service having reference to the source cluster it belongs to. Added.